domain/application/state/storage.go

```diff
@@ -843,16 +843,21 @@ func (st *State) makeInsertUnitStorageInstanceArgs(
 // makeInsertUnitStorageAttachmentArgs is responsible for making the set of
 // storage instance attachment arguments that correspond to the storage uuids.
 func makeInsertUnitStorageAttachmentArgs(
-	ctx context.Context,
+	_ context.Context,
 	unitUUID coreunit.UUID,
 	storageToAttach []domainstorage.StorageInstanceUUID,
 ) ([]insertStorageInstanceAttachment, error) {
 	rval := make([]insertStorageInstanceAttachment, 0, len(storageToAttach))
 	for _, instUUID := range storageToAttach {
+		saUUID, err := storageprovisioning.NewStorageAttachmentUUID()
```

Comment:
Can you please change the `CreateUnitStorageArg` structure so that for storage to attach we generate the attachment uuid in the service layer and supply it down to state?

This is inline with where we are supposed to be heading. I realise that volume and filesystem will still be violating this but they will get fixed up in the next few weeks with Harry's work around enablement through the provider to work on IAAS model's.